src/java/org/apache/cassandra/db/marshal/AbstractType.java

```diff
@@ -804,6 +892,18 @@ public ByteBuffer getMaskedValue()
         throw new UnsupportedOperationException("There isn't a defined masked value for type " + asCQL3Type());
     }
 
+    protected static <K, V extends AbstractType<?>> V getInstance(Map<K, V> instances, K key, Supplier<V> value)
+    {
+        V cached = instances.get(key);
+        if (cached != null)
+            return cached;
+
+        // We avoid constructor calls in Map#computeIfAbsent to avoid recursive update exceptions because the automatic
+        // fixing of subtypes done by the top-level constructor might attempt a recursive update to the instances map.
```

Comment:
If we are going to remove the automatic fixing of types done by DSE then this part is unnecessary.